src/Corax/Queries/UnaryMatch.Ordinal.cs

```diff
@@ -191,129 +435,147 @@ private static int FillFuncNumerical<TComparer>(ref UnaryMatch<TInner, TValueTyp
                     searcher, fieldId, value,
-                    &FillFuncSequence<LessThanOrEqualMatchComparer>, &AndWith,
-                    inner.Count, inner.Confidence.Min(QueryCountConfidence.Normal), distinct, take);
+                    mode == UnaryMatchOperationMode.Any ? &FillFuncSequenceAny<LessThanOrEqualMatchComparer> : &FillFuncSequenceAll<LessThanOrEqualMatchComparer>, &AndWith,
+                    inner.Count, inner.Confidence.Min(QueryCountConfidence.Normal), mode, take: take);
             }
             else
             {
                 return new UnaryMatch<TInner, TValueType>(
                     in inner, UnaryMatchOperation.LessThanOrEqual,
                     searcher, fieldId, value,
-                    &FillFuncNumerical<LessThanOrEqualMatchComparer>, &AndWith,
-                    inner.Count, inner.Confidence.Min(QueryCountConfidence.Normal), distinct, take);
+                    mode == UnaryMatchOperationMode.Any ? &FillFuncNumerical<LessThanOrEqualMatchComparer> : &FillFuncNumerical<LessThanOrEqualMatchComparer>, &AndWith,
+                    inner.Count, inner.Confidence.Min(QueryCountConfidence.Normal), mode, take: take);
             }
         }
 
-        public static UnaryMatch<TInner, TValueType> YieldNotEqualsMatch(in TInner inner, IndexSearcher searcher, int fieldId, TValueType value, bool distinct = true,
-            int take = -1)
+        public static UnaryMatch<TInner, TValueType> YieldNotEqualsMatch(in TInner inner, IndexSearcher searcher, int fieldId, TValueType value,
+            UnaryMatchOperationMode mode = UnaryMatchOperationMode.Any, int take = -1)
         {
             if (typeof(TValueType) == typeof(Slice))
             {
                 return new UnaryMatch<TInner, TValueType>(
                     in inner, UnaryMatchOperation.NotEquals,
                     searcher, fieldId, value,
-                    &FillFuncSequence<NotEqualsMatchComparer>, &AndWith,
-                    inner.Count, inner.Confidence.Min(QueryCountConfidence.Normal), distinct, take);
+                    mode == UnaryMatchOperationMode.Any ? &FillFuncSequenceAny<EqualsMatchComparer> : &FillFuncSequenceAll<EqualsMatchComparer>, &AndWith,
+                    inner.Count, inner.Confidence.Min(QueryCountConfidence.Normal), mode, true, take: take);
             }
             else
             {
                 return new UnaryMatch<TInner, TValueType>(
                     in inner, UnaryMatchOperation.NotEquals,
                     searcher, fieldId, value,
-                    &FillFuncNumerical<NotEqualsMatchComparer>, &AndWith,
-                    inner.Count, inner.Confidence.Min(QueryCountConfidence.Normal), distinct, take);
+                    mode == UnaryMatchOperationMode.Any ? &FillFuncNumerical<EqualsMatchComparer> : &FillFuncNumerical<EqualsMatchComparer>, &AndWith,
+                    inner.Count, inner.Confidence.Min(QueryCountConfidence.Normal), mode, true, take: take);
```

Comment:
You are passing `EqualsMatchComparer` in `NotEqualMatch` ;) 
@redknightlois

Reply:
This is about `QueryingOnEqNotNull`

Reply:
I am passing `EqualsMatchComparer` but I am also forcing `distinct = true` ... this is one of the side effects of `null` you have to be very careful with the `Any` and `All` distinction. It is not symetric as it was before... maybe the failed ones are related to `FillFuncNumerical` as I am seeing that they have no null handling.